src/cswinrt/strings/ComInteropHelpers.cs

```diff
@@ -485,3 +485,31 @@ public static DisplayInformation GetForMonitor(IntPtr monitor)
     }
 #endif
 }
+
+namespace Windows.ApplicationModel.DataTransfer
+{
+#if UAC_VERSION_1
+#if NET
+    [global::System.Runtime.Versioning.SupportedOSPlatform("Windows10.0.10240.0")]
+#endif
+#if EMBED
+    internal
+#else
+    public
+#endif
+    static class DataTransferManagerInterop
+    {
+        private static IDataTransferManagerInterop dataTransferManagerInterop = DataTransferManager.As<IDataTransferManagerInterop>();

```

Comment:
You shouldn't use `As<T>`, always pass an explicit IID to avoid rooting reflection metadata.

Reply:
> You shouldn't use `As<T>`, always pass an explicit IID to avoid rooting reflection metadata.

![image](https://github.com/microsoft/CsWinRT/assets/49179966/1432fa12-0d73-4239-bb7d-a772dcd92bc3)

At present, other similar interop apis do not find that they use other methods related to As<T>, do you have any good alternative ways to change all the interop apis?

Reply:
As far as I'm aware, the reason why these classes are using `As<T>` is because CsWinRT previously didn't have a way to easily get the ABI for activation factories, so this was done instead. This has been fixed with the `ActivationFactory` class in the AOT branch though.